### PR TITLE
Scala codegen StackOverflowError fix

### PR DESCRIPTION
**Pull request by leo-da**
More than 20 scala codegen runs on the DAR that caused problems, no SOE so far.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/dependencygraph/DependencyGraph.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ private final case class LFDependencyGraph(private val util: lf.LFUtil)
             symmGenTypeDependencies(typeDecl),
             collectDepError = false))
     }
-    val templateNodes = decls.collect {
+    val templateNodes = decls.toStream.collect {
```
Comment (S11001001) [inline language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/dependencygraph/DependencyGraph.scala]: Did you observe any value from these `toStream` calls? If not, I suggest reverting.

Comment (leo-da) [reply language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/dependencygraph/DependencyGraph.scala]: looks like it decreased the number of the SOE from 1 out 5 to 1 out 20.

**Comment by leo-da** [comment]
50 scripted scala codegen runs just completed on a very big DAR file without any errors:
```
#!/bin/bash -eux

max_num=50

[ -d ./tmp ] || mkdir ./tmp

for n in $(seq 1 1 $max_num); do
  echo "run $n"
  log="./tmp/codegen-scala-$n.log"

  touch app/daml-model/src/DA/<XXX>/Main/Common/Text.daml
  make daml-compile &> $log
done
```
The thread stack size was set to the default (corresponding config removed):
```
${SCALA_OUTPUT_DIR}: export JAVA_TOOL_OPTIONS = -Xmx4g
```
The default thread stack size on Linux is 1024 bytes BTW:
```
$ java -XX:+PrintFlagsFinal -version | grep " ThreadStackSize"
+ /home/leos/Projects/solution-spider/dev-env/bin/java -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/home/leos/tmp/scla-codegen.hprof -XX:+PrintFlagsFinal -version
     intx ThreadStackSize                           = 1024                                {pd product}
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
CC @jberthold-da

**Comment by jberthold-da** [comment]
:tada: thanks, looking good

**Pull request merged**